### PR TITLE
Pad confirmation related changes (#6)

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/5343

*Description of changes:*

* PAD confirmation period added

* Added the activate pad account tasks

* Added a template readme document hoping to put more details there

* Fixed failing tests and added more.

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updated the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/services/payment_service.py [hunk 1]
```diff
@@ -140,6 +142,14 @@ def create_invoice(cls, payment_request: Tuple[Dict[str, Any]], authorization: T
 
         return invoice.asdict()
 
+    @staticmethod
+    def _validate_and_throw_error(payment_method: str, payment_account: PaymentAccount):
```
Comment (saravanpa-aot) [inline pay-api/src/pay_api/services/payment_service.py:146]: is there a better place/where is the correct place to do this check? @sumesh-aot

File: pay-api/migrations/versions/b624b5dbc9b9_added_pad_activation_dates.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+# revision identifiers, used by Alembic.
+revision = 'b624b5dbc9b9'
+down_revision = '7c19ee3a58aa'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('cfs_account', sa.Column('pad_tos_accepted_by', sa.String(length=50), nullable=True))
```
Comment (sumesh-aot) [inline pay-api/migrations/versions/b624b5dbc9b9_added_pad_activation_dates.py:22]: Better to move this to payment_account table as new cfs_account record will be created whenever anything with CFS (bank details) changes.

File: pay-api/migrations/versions/b624b5dbc9b9_added_pad_activation_dates.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    error_code_table = table('error_code',
+                             column('code', String),
+                             column('title', String),
+                             column('detail', String)
+                             )
+    op.bulk_insert(
+        error_code_table,
+        [
+            {'code': 'ACCOUNT_IN_PAD_CONFIRMATION_PERIOD', 'title': 'Account in PAD confirmation period',
```
Comment (sumesh-aot) [inline pay-api/migrations/versions/b624b5dbc9b9_added_pad_activation_dates.py:40]: Please confirm this message with Amit.

File: pay-api/src/pay_api/services/payment_service.py (hunk 1 above)
Comment (sumesh-aot) [reply pay-api/src/pay_api/services/payment_service.py:146]: May be add this in the factory or pad_service invoice creation method ?

**Review by sumesh-aot — COMMENTED** [review]
Added some comments to consider

File: pay-api/migrations/versions/b624b5dbc9b9_added_pad_activation_dates.py (hunk 2 above)
Comment (saravanpa-aot) [reply pay-api/migrations/versions/b624b5dbc9b9_added_pad_activation_dates.py]: done

File: pay-api/migrations/versions/b624b5dbc9b9_added_pad_activation_dates.py (hunk 3 above)
Comment (saravanpa-aot) [reply pay-api/migrations/versions/b624b5dbc9b9_added_pad_activation_dates.py:40]: done

**Comment by saravanpa-aot** [comment]
changes addressed @sumesh-aot

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**